test/read_events.h

```diff
@@ -153,12 +157,67 @@ void processEvent(podio::EventStore& store, bool verboser, unsigned eventNum) {
     throw std::runtime_error("Collection 'SimCalorimeterHits' should be present");
   }
 
+  //===============================================================================
+  if( tpchs.isValid() ){
+
+    std::cout <<"\n collection: " << "Time Projection Chamber Hit" << " of type " << tpchs.getValueTypeName() << "\n\n"
+ 	      << tpchs << std::endl ;
+    int ntpch = 5 ;
+    for(int j=0 ; j<ntpch ; ++j){
+     
+      
+      auto tpch1 = tpchs[2*j];
+      if ( tpch1.getCellID() != 0xabadcaffee ) throw std::runtime_error("cellID != 0xabadcaffee") ;
+      if ( tpch1.getTime() !=  j*0.3f )  throw std::runtime_error("time != j*0.3") ;
+      if ( tpch1.getCharge() != j*2.f ) throw std::runtime_error("charge != j*2") ;
+
+      auto tpch2 = tpchs[2*j+1] ;
+      if ( tpch2.getCellID() != 0xcaffeebabe ) throw std::runtime_error("cellID != 0xcaffeebabe ") ;
+      if ( tpch2.getTime() !=  j*0.3f )  throw std::runtime_error("time != j*0.3") ;
+      if ( tpch2.getCharge() != -j*2.f ) throw std::runtime_error("charge != -j*2") ;
+    }
+
+  }else {
+      throw std::runtime_error("Collection 'TCPHits' should be present");
+    }
+
+ //===============================================================================
+ if( thps.isValid() ){
+   std::cout <<"\n collection: " << "Tracker Hit Plane" << " of type " << thps.getValueTypeName() << "\n\n"
+ 	      << thps << std::endl ;
+   int nthp = 5 ;
+   for(int j=0 ; j<nthp ; ++j){
+     auto thp1 = thps[2*j];
+     
+     if ( thp1.getCellID() != 0xabadcaffee ) throw std::runtime_error("cellID != 0xabadcaffee") ;
+     if ( thp1.getTime() !=  j*0.3f )  throw std::runtime_error("time != j*0.3") ;
+     if ( thp1.getEDep() !=  j * 0.000001f )  throw std::runtime_error("eDep != j * 0.000001") ;
+     if( !( thp1.getPosition() == edm4hep::Vector3d( j*2. , j*3. , j*5.  )) )
+          throw std::runtime_error("position != ( j*2. , j*3. , j*5. )") ;
+
+     auto thp2 = thps[2*j+1];
+     
+     if ( thp2.getCellID() != 0xcaffeebabe ) throw std::runtime_error("cellID != 0xcaffeebabe") ;
+     if ( thp2.getTime() !=  j*0.3f )  throw std::runtime_error("time != j*0.3") ;
+     if ( thp2.getEDep() !=  j * 0.0000031f )  throw std::runtime_error("eDep != j * 0.0000031 ") ;
+     if( !( thp2.getPosition() == edm4hep::Vector3d( -j*2. , -j*3. , -j*5.  )) )
+          throw std::runtime_error("position != ( -j*2. , -j*3. , -j*5. )") ;
+   }
+ }else {
+      throw std::runtime_error("Collection 'TrackerHitPlanes' should be present");
+ }
+
  // //===============================================================================
  //  if( sccons.isValid() ){
  //  } else {
  //    throw std::runtime_error("Collection 'SimCalorimeterHitContributions' should be present");
  //  }
 
+auto& evtMD = store.getEventMetaData() ;
+   std::string evtType = evtMD.getStringVal( "EventType" ) ;
```

Comment:
```suggestion
   const auto& evtType = evtMD.getValue<std::string>( "EventType" ) ;
```
Templated access has been added in AIDASoft/podio#262 . In the course of that this interface has been marked as deprecated. If you build against a recent enough podio version you should be able to pick up the newer interface.